ruby/example_code/lambda/lambda_basics.rb

```diff
@@ -0,0 +1,295 @@
+        function_name: function_name
+      }
+    )
+  rescue Aws::Lambda::Errors::ResourceNotFoundException => e
+    @logger.debug("Could not find function: #{function_name}:\n #{e.message}")
+    nil
+  end
+  # snippet-end:[ruby.example_code.lambda.GetFunction]
+
+  # snippet-start:[ruby.example_code.lambda.CreateFunction]
+  # Deploys a Lambda function.
+  #
+  # @param function_name: The name of the Lambda function.
+  # @param handler_name: The fully qualified name of the handler function. This
+  #                      must include the file name and the function name.
+  # @param role_arn: The IAM role to use for the function.
+  # @param deployment_package: The deployment package that contains the function
+  #                            code in .zip format.
+  # @return: The Amazon Resource Name (ARN) of the newly created function.
+  def create_function(function_name, handler_name, role_arn, deployment_package)
+    response = @lambda_client.create_function({
+                                                role: role_arn.to_s,
+                                                function_name: function_name,
+                                                handler: handler_name,
+                                                runtime: "ruby2.7",
+                                                code: {
+                                                  zip_file: deployment_package
+                                                },
+                                                environment: {
+                                                  variables: {
+                                                    "LOG_LEVEL" => "info"
+                                                  }
+                                                }
+                                              })
+    @lambda_client.wait_until(:function_active_v2, { function_name: function_name }) do |w|
+      w.max_attempts = 5
+      w.delay = 5
+    end
+    response
+  rescue Aws::Lambda::Errors::ServiceException => e
+    @logger.error("There was an error creating #{function_name}:\n #{e.message}")
+  rescue Aws::Waiters::Errors::WaiterFailed => e
+    @logger.error("Failed waiting for #{function_name} to activate:\n #{e.message}")
+  end
+  # snippet-end:[ruby.example_code.lambda.CreateFunction]
+
+  # snippet-start:[ruby.example_code.lambda.Invoke]
+  # Invokes a Lambda function.
+  # @param function_name: The name of the function to invoke.
+  # @return: The response from the function invocation.
+  # @param [nil] payload
+  def invoke_function(function_name, payload = nil)
+    params = { function_name: function_name }
+    params[:payload] = payload unless payload.nil?
+    @lambda_client.invoke(params)
+  rescue Aws::Lambda::Errors::ServiceException => e
+    @logger.error("There was an error executing #{function_name}:\n #{e.message}")
+  end
+  # snippet-end:[ruby.example_code.lambda.Invoke]
+
+  # Get function logs from the latest CloudWatch log stream
```

Comment:
Get function logs from the latest Amazon CloudWatch Logs stream.

Reply:
Done.